controllers/envoyfleet_resources.go

```diff
@@ -0,0 +1,320 @@
+		labels[key] = value
+	}
+
+	configMapName := "kusk-gateway-envoy-config-" + e.fleet.Name
+
+	xdsLabels := map[string]string{"app.kubernetes.io/name": "kusk-gateway", "app.kubernetes.io/component": "xds-service"}
+	xdsServices, err := k8sutils.GetServicesByLabels(ctx, e.client, xdsLabels)
+	if err != nil {
+		return fmt.Errorf("cannot create Envoy Fleet %s config map: %w", e.fleet.Name, err)
+	}
+	switch svcs := len(xdsServices); {
+	case svcs == 0:
+		return fmt.Errorf("cannot create Envoy Fleet %s config map: no xds services detected in the cluster when searching with the labels %s", e.fleet.Name, xdsLabels)
+	case svcs > 1:
+		return fmt.Errorf("cannot create Envoy Fleet %s config map: multiple xds services detected in the cluster when searching with the labels %s", e.fleet.Name, xdsLabels)
+	}
+	// At this point - we have exactly one service with (we ASSUME!) one port
+	xdsServiceHostname := fmt.Sprintf("%s.%s.svc.cluster.local.", xdsServices[0].Name, xdsServices[0].Namespace)
+	xdsServicePort := xdsServices[0].Spec.Ports[0].Port
+
+	e.configMap = &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            configMapName,
+			Namespace:       e.fleet.Namespace,
+			Labels:          labels,
+			OwnerReferences: []metav1.OwnerReference{envoyFleetAsOwner(e.fleet)},
+		},
+		Data: map[string]string{
+			"envoy-config.yaml": fmt.Sprintf(envoyConfigTemplate, e.fleet.Name, xdsServiceHostname, xdsServicePort),
+		},
+	}
+
+	return nil
+}
+
+func (e *EnvoyFleetResources) generateDeployment() {
+	// future object labels
+	labels := map[string]string{
+		"app.kubernetes.io/component": "envoy",
+	}
+	// Copy over shared labels map
+	for key, value := range e.sharedLabels {
+		labels[key] = value
+	}
+
+	deploymentName := "kusk-gateway-envoy-" + e.fleet.Name
+
+	configMapName := e.configMap.Name
+
+	// Create container template first
+	envoyContainer := corev1.Container{
+		Name:            "envoy",
+		Image:           e.fleet.Spec.Image,
+		ImagePullPolicy: "IfNotPresent",
+		Command:         []string{"/bin/sh", "-c"},
+		Args: []string{
+			"envoy -c /etc/envoy/envoy.yaml --service-node $POD_NAME",
+		},
+		Env: []corev1.EnvVar{
+			{
+				Name: "POD_NAME",
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						FieldPath: "metadata.name",
+					},
+				},
+			},
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      "envoy-config",
+				MountPath: "/etc/envoy/envoy.yaml",
+				SubPath:   "envoy-config.yaml",
+			},
+		},
+		Ports: []corev1.ContainerPort{
+			{
+				Name:          "http",
+				ContainerPort: envoyHTTPListenerPort,
+			},
+			{
+				Name:          "https",
+				ContainerPort: envoyHTTPSListenerPort,
+			},
+			{
+				Name:          "admin",
+				ContainerPort: envoyAdminListenerPort,
+			},
+		},
+	}
+	// Set Enovy Pod Resources if specified
+	if e.fleet.Spec.Resources != nil {
```

Comment:
Should we have some sensible defaults here for limits or requests if either one is not defined?

Reply:
The recommended way is to set this via Helm or any other deployment. Helm developers don't recommend to set this too and depend on the user needs.